### PR TITLE
Record how to mark a test as flaky

### PR DESCRIPTION
**Pull request by xdje42**
Adds instructions for how to mark flaky tests.
Tested by viewing $build/api/docs/embed/html/page_test_suite.html.

### CODE REVIEW

File: api/docs/test_suite.dox [hunk 1]
```diff
@@ -315,6 +315,15 @@ The comments at the top of runsuite_ssh.cmake describe additional options.
```
Comment (derekbruening) [inline api/docs/test_suite.dox:315]: > iX: Record how to mark a test as flaky

Repeating the same comment from your first PR: we do not put "IX: " in titles; there is no prefix if there is no issue.

File: api/docs/test_suite.dox [hunk 2]
```diff
@@ -315,6 +315,15 @@ The comments at the top of runsuite_ssh.cmake describe additional options.
 
 Unfortunately our test suite is not as clean as it could be.  Some tests can be flaky and while they pass on the machines of the existing developers and on our automated test machines, they may fail occasionally on a new machine.  Please search the issue tracker before filing a new issue to see if a test failure has already been seen once before.  We welcome contributions to fix flaky tests.
 
+Flaky tests are marked in one of two ways:
+
+ - Append "_FLAKY" to the test's name
+    - See suite/tests/CMakeLists.txt for examples
```
Comment (derekbruening) [inline api/docs/test_suite.dox:321]: Let's make these links to the online file viewer, here and below.

File: api/docs/test_suite.dox [hunk 3]
```diff
@@ -315,6 +315,15 @@ The comments at the top of runsuite_ssh.cmake describe additional options.
 
 Unfortunately our test suite is not as clean as it could be.  Some tests can be flaky and while they pass on the machines of the existing developers and on our automated test machines, they may fail occasionally on a new machine.  Please search the issue tracker before filing a new issue to see if a test failure has already been seen once before.  We welcome contributions to fix flaky tests.
 
+Flaky tests are marked in one of two ways:
```
Comment (derekbruening) [inline api/docs/test_suite.dox:318]: If we're updating docs about flaky tests, let's talk about the SKIP_FLAKY_TESTS option here.

Comment (derekbruening) [inline api/docs/test_suite.dox:318]: Let's update the runsuite.cmake comments (line 141) about _FLAKY as well as they are stale.

File: api/docs/test_suite.dox [hunk 4]
```diff
@@ -315,6 +315,15 @@ The comments at the top of runsuite_ssh.cmake describe additional options.
...
+
+ - Append "_FLAKY" to the test's name
+    - See suite/tests/CMakeLists.txt for examples
+ - Mention the test in runsuite_wrapper.pl
```
Comment (derekbruening) [inline api/docs/test_suite.dox:322]: Re-iterate that an issue should always be filed covering fixing the test, and its issue number should be referenced when added here.

File: api/docs/test_suite.dox (hunk 3 above)
Comment (xdje42) [reply api/docs/test_suite.dox:318]: I propose doing this in a separate patch as it's orthogonal to marking tests.

Comment (xdje42) [reply api/docs/test_suite.dox:318]: What's the right wording to use here?

The code refers to #2974 but that seems to be just a general bug for recording the addition of aarch64 support.

Here's the best I can come up with:
```
Skip tests marked _FLAKY. This is one of the ways that flaky tests are marked as such
(the other way, for higher priority flakes to fix, is to mention them in runsuite_wrapper.pl).
We need CDash to ignore flakes and avoid going red and sending emails,
and we rely on our CI for a history of _FLAKY results.
```

File: api/docs/test_suite.dox (hunk 2 above)
Comment (xdje42) [reply api/docs/test_suite.dox]: Done

File: api/docs/test_suite.dox (hunk 4 above)
Comment (xdje42) [reply api/docs/test_suite.dox:322]: Done

File: api/docs/test_suite.dox (hunk 1 above)
Comment (xdje42) [reply api/docs/test_suite.dox:315]: Done, with separate PR #6586 to codify the rule.

File: api/docs/test_suite.dox (hunk 3 above)
Comment (derekbruening) [reply api/docs/test_suite.dox:318]: We can remove the CDash sentence as we no longer use CDash.

**Review by derekbruening — APPROVED** [review]

File: api/docs/test_suite.dox (hunk 3 above)
Comment (xdje42) [reply api/docs/test_suite.dox:318]: Done

**Pull request merged**